azurerm/internal/services/network/resource_arm_application_gateway.go

```diff
@@ -1430,6 +1436,14 @@ func resourceArmApplicationGatewayCreateUpdate(d *schema.ResourceData, meta inte
 		gateway.ApplicationGatewayPropertiesFormat.WebApplicationFirewallConfiguration = expandApplicationGatewayWafConfig(d)
 	}
 
+	if res, ok := d.GetOk("firewall_policy_id"); ok {
+		vs := res.([]interface{})
+		id := vs[0].(string)
+		gateway.ApplicationGatewayPropertiesFormat.FirewallPolicy = &network.SubResource{
+			ID: &id,
+		}
+	}
```

Comment:
I think this can/should be
```suggestion
	if v, ok := d.GetOk("firewall_policy_id"); ok {
		gateway.ApplicationGatewayPropertiesFormat.FirewallPolicy = &network.SubResource{
			ID: &utils.String(v.(string)),
		}
	}
```
?